repl/evaluator.go

```diff
@@ -122,22 +126,33 @@ func (e *Evaluator) updateContextPlans(ctx *EvaluationContext) error {
 	env := e.env
 	for i := range ctx.letVars {
 		el := &ctx.letVars[i]
-		if el.prog == nil {
+		if el.prog == nil && el.src != "" {
```

Comment:
A quick comment above the block to indicate what this condition indicates would be good. Does this condition relate in any way to the `applyContext()` condition used later on?